src/codegen/general.rs

```diff
@@ -143,6 +148,9 @@ pub fn define_shared_type(
     try!(writeln!(w, "\tmatch fn {{"));
     try!(writeln!(w, "\t\tref => |ptr| ffi::{}(ptr),", ref_fn));
     try!(writeln!(w, "\t\tunref => |ptr| ffi::{}(ptr),", unref_fn));
+    if let &Some(ref get_type_fn) = get_type_fn {
```

Comment:
Maybe better `Some() = *get_type_fn()`
And in `codegen\translate_from_glib.rs:34:25 if let &Some(_) = array_length` please too .